### PR TITLE
Lims 4281/4297

### PR DESCRIPTION
**Pull request by dianaeldemerdash**

### CODE REVIEW

File: ui_testing/testcases/basic_tests/test03_orders.py [hunk 1]
```diff
@@ -112,6 +112,90 @@ def test003_cancel_button_edit_departments(self, save):
                                                                                       order_departments))
             self.assertEqual(current_departments, order_departments)
 
+    @parameterized.expand(['save_btn', 'cancel'])
+    def test006_update_material_type(self, save):
```
Comment (0xIslamTaha) [inline ui_testing/testcases/basic_tests/test03_orders.py]: add new test cases to the end of the file

File: ui_testing/testcases/basic_tests/test03_orders.py [hunk 2]
```diff
@@ -112,6 +112,90 @@ def test003_cancel_button_edit_departments(self, save):
...
+
+        LIMS-4281
+        LIMS-4281
+        :return:
+        """
+        self.order_page.get_random_order()
+        order_url = self.base_selenium.get_url()
+        self.base_selenium.LOGGER.info(' + order_url : {}'.format(order_url))
+        current_material_type = self.order_page.get_material_type()
+        self.order_page.set_material_type()
```
Comment (0xIslamTaha) [inline ui_testing/testcases/basic_tests/test03_orders.py]: changing material type raises a pop-up and you didn't handle it.
image

File: ui_testing/testcases/basic_tests/test03_orders.py [hunk 3]
```diff
@@ -112,6 +112,90 @@ def test003_cancel_button_edit_departments(self, save):
...
+        new_material_type = self.order_page.get_material_type()
```
Comment (0xIslamTaha) [inline ui_testing/testcases/basic_tests/test03_orders.py]: set_material_type with random value returns the new value, so u dont need to get it again

File: ui_testing/testcases/basic_tests/test03_orders.py [hunk 4]
```diff
@@ -112,6 +112,90 @@ def test003_cancel_button_edit_departments(self, save):
...
+        self.order_page.confirm_popup(force=True)
```
Comment (0xIslamTaha) [inline ui_testing/testcases/basic_tests/test03_orders.py]: this should be before get_material_type

File: ui_testing/testcases/basic_tests/test03_orders.py [hunk 5]
```diff
@@ -112,6 +112,90 @@ def test003_cancel_button_edit_departments(self, save):
...
+        self.order_page.get_article()
+        self.order_page.set_article()
```
Comment (0xIslamTaha) [inline ui_testing/testcases/basic_tests/test03_orders.py]: setting a general article may be lead to `empty` article.

File: ui_testing/testcases/basic_tests/test03_orders.py [hunk 6]
```diff
@@ -112,6 +112,90 @@ def test003_cancel_button_edit_departments(self, save):
...
+        self.order_page.get_test_plan()
+        self.order_page.set_test_plan()
```
Comment (0xIslamTaha) [inline ui_testing/testcases/basic_tests/test03_orders.py]: setting a general test_plan may be lead to `empty` article as u can choose material_type+article which doesn't have a test plan.
image

File: ui_testing/testcases/basic_tests/test03_orders.py (hunk 1 above)
Comment (dianaeldemerdash) [reply ui_testing/testcases/basic_tests/test03_orders.py]: What do you mean by that ?

Comment (dianaeldemerdash) [reply ui_testing/testcases/basic_tests/test03_orders.py]: What do you mean by that ?

File: ui_testing/testcases/basic_tests/test03_orders.py (hunk 2 above)
Comment (dianaeldemerdash) [reply ui_testing/testcases/basic_tests/test03_orders.py]: I already put  ( self.order_page.confirm_popup(force=True) )

File: ui_testing/testcases/basic_tests/test03_orders.py (hunk 4 above)
Comment (dianaeldemerdash) [reply ui_testing/testcases/basic_tests/test03_orders.py]: Done

File: ui_testing/testcases/basic_tests/test03_orders.py (hunk 3 above)
Comment (dianaeldemerdash) [reply ui_testing/testcases/basic_tests/test03_orders.py]: You mean we don't need to ( new_material_type = self.order_page.get_material_type())

File: ui_testing/testcases/basic_tests/test03_orders.py (hunk 5 above)
Comment (dianaeldemerdash) [reply ui_testing/testcases/basic_tests/test03_orders.py]: I did a solution get article only without set it ( I mean when i'm in suborders view I want to make sure that this article that I updated according to the materiel type it updated successfully in the suborder view )

File: ui_testing/testcases/basic_tests/test03_orders.py (hunk 6 above)
Comment (dianaeldemerdash) [reply ui_testing/testcases/basic_tests/test03_orders.py]: What about to give a test plan name in set_test_plan?? or get test plan only without set it

File: ui_testing/testcases/basic_tests/test03_orders.py (hunk 1 above)
Comment (0xIslamTaha) [reply ui_testing/testcases/basic_tests/test03_orders.py]: new test cases should be added to the end of the tests file not in the middle of the code.

it should go after this line
image

File: ui_testing/testcases/basic_tests/test03_orders.py (hunk 2 above)
Comment (0xIslamTaha) [reply ui_testing/testcases/basic_tests/test03_orders.py]: u put it in a wrong order. Execute your test case and I will see that it doesnot interact with the pop-up

File: ui_testing/testcases/basic_tests/test03_orders.py (hunk 3 above)
Comment (0xIslamTaha) [reply ui_testing/testcases/basic_tests/test03_orders.py]: yes

File: ui_testing/testcases/basic_tests/test03_orders.py (hunk 5 above)
Comment (0xIslamTaha) [reply ui_testing/testcases/basic_tests/test03_orders.py]: I already made it clear in set_test_plan() comment. So the solution is choosing a specific material type+active_article+test_plan to b valid

File: ui_testing/testcases/basic_tests/test03_orders.py (hunk 6 above)
Comment (0xIslamTaha) [reply ui_testing/testcases/basic_tests/test03_orders.py]: if u change the Material type then you need to choose a related active article and test plan so I think u need to `set` it. As I said in the previouse comment,  So the solution is choosing a specific material type+active_article+test_plan to b valid

File: ui_testing/testcases/basic_tests/test03_orders.py (hunk 1 above)
Comment (dianaeldemerdash) [reply ui_testing/testcases/basic_tests/test03_orders.py]: Done

File: ui_testing/testcases/basic_tests/test03_orders.py (hunk 2 above)
Comment (dianaeldemerdash) [reply ui_testing/testcases/basic_tests/test03_orders.py]: This is should put before get ( done)

File: ui_testing/testcases/basic_tests/test03_orders.py (hunk 3 above)
Comment (dianaeldemerdash) [reply ui_testing/testcases/basic_tests/test03_orders.py]: Done

**Pull request closed without merging**